src/cluster/src/sys.rs

```diff
@@ -0,0 +1,223 @@
+use tracing::{info, debug, instrument};
+use derive_builder::Builder;
+use crate::{
+    ChartLocation, DEFAULT_CHART_APP_REPO, DEFAULT_CHART_SYS_REPO, DEFAULT_NAMESPACE,
+    DEFAULT_CHART_REMOTE,
+};
+use fluvio_helm::{HelmClient, InstallArg};
+use std::path::{Path, PathBuf};
+use crate::error::SysInstallError;
+
+const DEFAULT_SYS_NAME: &str = "fluvio-sys";
+const DEFAULT_CHART_SYS_NAME: &str = "fluvio/fluvio-sys";
+const DEFAULT_CLOUD_NAME: &str = "minikube";
+
+/// Configuration options for installing Fluvio system charts
+#[derive(Builder, Debug)]
+#[builder(build_fn(skip), setter(prefix = "with"))]
+pub struct SysConfig {
+    /// The type of cloud infrastructure the cluster will be running on
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use fluvio_cluster::SysConfigBuilder;
+    /// # fn add_cloud(builder: &mut SysConfigBuilder) {
+    /// builder.with_cloud("minikube");
+    /// # }
+    /// ```
+    #[builder(setter(into))]
+    pub cloud: String,
+    /// The namespace in which to install the system chart
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use fluvio_cluster::SysConfigBuilder;
+    /// # fn add_namespace(builder: &mut SysConfigBuilder) {
+    /// builder.with_namespace("fluvio");
+    /// # }
+    /// ```
+    #[builder(setter(into))]
+    pub namespace: String,
+    /// The location at which to find the system chart to install
+    pub chart_location: ChartLocation,
+}
+
+impl SysConfig {
+    /// Creates a default [`SysConfigBuilder`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use fluvio_cluster::SysConfig;
+    /// let builder = SysConfig::builder();
+    /// ```
+    pub fn builder() -> SysConfigBuilder {
+        SysConfigBuilder::default()
+    }
+}
+
+impl SysConfigBuilder {
+    /// Validates all builder options and constructs a `SysConfig`
+    pub fn build(&self) -> Result<SysConfig, SysInstallError> {
```

Comment:
Shouldn't this just return `SysConfig` rater than Result?  Why would this fail?

Reply:
This is intentional, the reason I put a Result there is because even though there might not be a reason for this to fail today, in the future we may need to add configuration options that require validation or which need to be collected from the environment, which may be fallible operations. It doesn't hurt us today to put a Result here if we don't need it, but it may hurt us in the future (by introducing a breaking change) if we end up needing a Result but don't have one.